main.py

```diff
@@ -2,19 +2,57 @@
 
 import argparse
 
+from shapely.geometry import MultiPolygon, Polygon
 from sqlalchemy import and_, func, select
 from sqlalchemy.exc import SQLAlchemyError
 
 from checks import autocomplete_check, check_topmost
 from config import Config
-from google_api_calls import text_search
+from google_api_calls import geocoding_api, text_search
 from models import GivingPartnerLocations as gpl
 from models import GivingPartners as gp
 from models import get_engine, get_session
 
 logger = Config.logger
 
 
+def reverse_coordinates(coordinate_pairs):
+    """Function to reverse convert coordinate format from [long, lat] to [lat, long] to help support correct MySQL insertion of the building outline polygons"""  # pylint: disable=line-too-long
+    reversed_list = []
+    for coordinates in coordinate_pairs:
+        reversed_list.append(reversed(coordinates))
+    return reversed_list
+
+
+def preprocess_building_outlines(building_outlines):
+    """Returns the building outlines co-ordinates as a geometry object"""
+    shapely_geometry = None
+    if building_outlines and len(building_outlines) > 0:
+        coordinates = building_outlines["coordinates"]
+        try:
+            if building_outlines["type"].lower() == "polygon":
+                reversed_coordinates = reverse_coordinates(coordinates[0])
+                shapely_geometry = Polygon(reversed_coordinates)
+            elif building_outlines["type"].lower() == "multipolygon":
+                polygons = []
+                for polygon_coords in coordinates:
+                    reversed_coordinates = reverse_coordinates(polygon_coords[0])
+                    exterior_ring = reversed_coordinates
+                    interior_rings = (
+                        reversed_coordinates[1:] if len(polygon_coords) > 1 else None
+                    )
+                    polygons.append(Polygon(exterior_ring, interior_rings))
+
+                shapely_geometry = MultiPolygon(polygons)
+        except (ValueError, TypeError) as e:
+            logger.error(e)
+            raise e
+
+    if shapely_geometry is not None:
```

Comment:
```suggestion
    if shapely_geometry:
```
None in python is falsy